src/private/appEntity.ts

```diff
@@ -0,0 +1,73 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts, SdkError } from '../public';
+
+/**
+ * Namespace to interact with the application entities specific part of the SDK.
+ *
+ * @private
+ * Hide from docs
+ */
+export namespace appEntity {
+  /**
+   * @private
+   * Hide from docs
+   * --------
+   * Information on an app entity
+   */
+  export interface AppEntityConfiguration {
+    /**
+     * App ID of the application
+     */
+    appId: string;
+
+    /**
+     * URL for the application's icon
+     */
+    appIconUrl: string;
+
+    /**
+     * Content URL for the app entity
+     */
+    contentUrl: string;
+
+    /**
+     * The display name for the app entity
+     */
+    displayName: string;
+
+    /**
+     * Website URL for the app entity. It is meant to be opened by the user in a browser.
+     */
+    websiteUrl: string;
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Open the Tab Gallery and retrieve the app entity configuration
+   * @param channelId ID of the channel where the app entity will be created
+   * @param categories A list of app categories that will be displayed in the open tab gallery
+   * @param callback Callback that will be triggered once the app entity information is available.
+   *                 The callback takes two arguments: the app entity configuration, if available and
+   *                 an optional SdkError in case something happened (i.e. the window was closed)
+   */
+  export function selectAppEntity(
+    channelId: string,
```

Comment:
I wonder if we should just call this `threadId` since we might want to expand this functionality to chats/meetings at some point...